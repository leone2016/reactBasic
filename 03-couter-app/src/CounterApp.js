import React, { useState } from 'react';
import PropTypes from 'prop-types';
const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    //handleAdd
    const handleAdd = (operation) => {
          setCounter(counter + 1);
        //setCounter( (c)=> c+1);
    }
    const handlerReset = () => {
          setCounter(value);
    }
    const handlerSubs = () => {
        setCounter(counter - 1);
  }
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handlerReset}>Reset</button>
            <button onClick={handlerSubs}>-1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
export default CounterApp;