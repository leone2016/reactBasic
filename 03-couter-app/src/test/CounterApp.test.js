/**
 * tarea 
 * 1. Crear las siguientes pruebas para el <CounterApp/>
 *    Debe mostrar <CounterApp/> correctamente (hacer match con un snapshot)
 *    y sus valores por defecto
 * 
 * 2. Debe mostrar el valor por defecto de 100
 *    usar el wrapper.find, tomando el elemento html donde se muestra el valor del contador
 */
import React from 'react';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('UNIT TEST CounterApp.test.js', () => {
    let wrapper = shallow(<CounterApp />); // we define twice because when we init here the wrapper's properties are accesable in the test suit

    beforeEach(() => {
        console.log('BEFORE---------------------->');
        wrapper = shallow(<CounterApp />);
    })
    test('should be show <CounterApp/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be show the value by default 100', () => {
        const wrapper = shallow(<CounterApp value={100} />);
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('100');
    });

    test('should be increment with the button +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('11');
    });

    test('should be decrementar with the button -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('9');
    });

    test('should be reset the value with the default value ', () => {
        const wrapper = shallow(<CounterApp value={100} />);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('100');
    })
    
    

})
