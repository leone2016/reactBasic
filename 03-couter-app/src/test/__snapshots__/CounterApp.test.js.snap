// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UNIT TEST CounterApp.test.js should be show <CounterApp/> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
